tutorials/preprocessing/plot_70_fnirs_processing.py

```diff
@@ -0,0 +1,199 @@
+"""
+.. _tut-fnirs-processing:
+
+Preprocessing fNIRS data
+========================
+
+This tutorial covers how to convert fNIRS data from raw measurements to
+relative oxyhaemoglobin (HbO) and deoxyhaemoglobin (HbR) concentration.
+
+.. contents:: Page contents
+   :local:
+   :depth: 2
+
+Here we will work with the :ref:`fNIRS motor data <fnirs-motor-dataset>`.
+"""
+# sphinx_gallery_thumbnail_number = 3
+
+import os
+import mne
+from itertools import compress
+import matplotlib.pyplot as plt
+from mne.io.pick import _picks_to_idx
+
+
+fnirs_data_folder = mne.datasets.fnirs_motor.data_path()
+fnirs_raw_dir = os.path.join(fnirs_data_folder, 'Participant-1')
+raw_intensity = mne.io.read_raw_nirx(fnirs_raw_dir, verbose=True).load_data()
+
+
+###############################################################################
+# Selecting channels appropriate for detecting neural responses
+# -------------------------------------------------------------
+#
+# First we select optodes that are sufficiently separated to detect neural
+# responses (approximately 3cm) for further processing.
+
+long_channels = ~mne.preprocessing._beer_lambert_law._short_channels(
+    raw_intensity)
+raw_intensity.pick(mne.pick_channels(raw_intensity.ch_names,
+                                     list(compress(raw_intensity.ch_names,
```

Comment:
@larsoner we're not getting crossref link to `intertools.compress` here. Should we add `itertools` to intersphinx?

Reply:
It's already in `intersphinx` because `itertools` is a built-in module.

Doing `python -m sphinx.ext.intersphinx https://docs.python.org/3.7/objects.inv > ~/Desktop/python.txt` I can see it's under the `py:function` list, so not sure why it doesn't show up. It's a SG bug and I've opened a PR to fix it:

https://github.com/sphinx-gallery/sphinx-gallery/pull/558